### PR TITLE
Module 2 - Added devise, model User, main views in slim

### PR DESCRIPTION
**Pull request by sinpai**
Things done in this PR:
* Notes from mentor from previous PR has been fixed
* Devise has been installed
* Model User has been added
* Views in Slim for common devise actions has been added
* Main navigation and navbar has been added

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable, :trackable
```
Comment (AleksSenkou) [inline app/models/user.rb]: @sinpai as we discussed before: no need for comments if it's not a hacky logic, which you are trying to explain other developers

File: app/views/devise/confirmations/new.html.slim [hunk 2]
```diff
@@ -0,0 +1,14 @@
+= bootstrap_devise_error_messages!
+.row
+  .col-md-6.offset-md-3
+    .panel.panel-default.devise-bs
+      .panel-heading.text-center
+        h4
+          = t('.resend_confirmation_instructions', default: 'Resend confirmation instructions')
```
Comment (AleksSenkou) [inline app/views/devise/confirmations/new.html.slim]: @sinpai no need in `:default` option, let's just keep locales up to date.

In order to add locale [check this article](http://guides.rubyonrails.org/i18n.html)

I've noticed it in several places more. So, be sure to remove this `default` option keys everywhere and add needed locales to `en.yml`

File: bin/webpack [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 require 'pathname'
 ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../../Gemfile',
-  Pathname.new(__FILE__).realpath)
+                                           Pathname.new(__FILE__).realpath)
```
Comment (AleksSenkou) [inline bin/webpack]: oh 😄

File: config/environments/development.rb [hunk 4]
```diff
@@ -30,6 +30,7 @@
   end
 
   # Don't care if the mailer can't send.
+  config.action_mailer.default_url_options = {host: 'localhost', port: 3000}
```
Comment (AleksSenkou) [inline config/environments/development.rb]: @sinpai let's not save space if it's not needed:
```ruby
config.action_mailer.default_url_options = {
  host: 'localhost',
  port: 3000
}
```

Comment (AleksSenkou) [reply config/environments/development.rb]: @sinpai by the way: what do you think about moving `3000` to a ENV variable? you might use `3000` in sevaral places across the app more, so you'll need to change value only in one place

File: config/environments/production.rb [hunk 5]
```diff
@@ -60,6 +60,8 @@
   # Use a different cache store in production.
   # config.cache_store = :mem_cache_store
 
+  config.action_mailer.default_url_options = {host: 'datarockets-finance.herokuapp.com', port: 3000}
```
Comment (AleksSenkou) [inline config/environments/production.rb]: @sinpai the same thing here

File: config/initializers/devise.rb [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+Devise.setup do |config|
+  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
+  require 'devise/orm/active_record'
```
Comment (AleksSenkou) [inline config/initializers/devise.rb]: @sinpai why this requirement goes here but not at the start of the file?

File: config/linters/reek.yml [hunk 7]
```diff
@@ -1,4 +1,4 @@
 IrresponsibleModule:
   enabled: false
 NestedIterators:
-  max_allowed_nesting: 2
\ No newline at end of file
+  max_allowed_nesting: 2
```
Comment (AleksSenkou) [inline config/linters/reek.yml:4]: 👍

**Review by AleksSenkou — COMMENTED** [review]
@sinpai well done and great PR description 🥇 

> Notes from mentor from the previous PR has been fixed

Let's fix comment in a PR where they were left in future.

File: app/models/user.rb (hunk 1 above)
Comment (sinpai) [reply app/models/user.rb]: :+1:

File: app/views/devise/confirmations/new.html.slim (hunk 2 above)
Comment (sinpai) [reply app/views/devise/confirmations/new.html.slim]: :+1:

File: config/environments/development.rb (hunk 4 above)
Comment (sinpai) [reply config/environments/development.rb]: 👍 moved to DEFAULT_PORT variable

File: bin/webpack (hunk 3 above)
Comment (sinpai) [reply bin/webpack]: 😄

File: app/views/home_page/show.html.slim [hunk 8]
```diff
@@ -4,4 +4,5 @@ div.jumbotron
   hr.my-4
   p For now it's just a simple page
   p.lead
-    a class=['btn','btn-primary','btn-lg'] role="button" Learn more
+    a.btn.btn-primary.btn-lg[role="button"]
+      | Learn more
```
Comment (AleksSenkou) [inline app/views/home_page/show.html.slim:8]: @sinpai `Learn more` should be moved to locales as other plain text 😉

File: app/views/shared/_navbar.html.slim [hunk 9]
```diff
@@ -0,0 +1,8 @@
+nav.navbar.navbar-toggleable-md.navbar-inverse.bg-primary
+  .container
+    button.navbar-toggler.navbar-toggler-right[type="button" data-toggle="collapse" data-target="#navbarNavAltMarkup" aria-controls="navbarNavAltMarkup" aria-expanded="false" aria-label="Toggle navigation"]
+      span.navbar-toggler-icon
+    a.navbar-brand[href="/"] Personal Finance App
```
Comment (AleksSenkou) [inline app/views/shared/_navbar.html.slim:5]: @sinpai `Personal Finance App` to locale

**Review by AleksSenkou — APPROVED** [review]
LGTM with suggestions 🥇

@sinpai check your views for plain text. If you notice any, be sure to move them to locales (devise.yml, en.yml)

**Comment by AleksSenkou** [comment]
@sinpai oh, I see you moved those to locales in the next PR 👍 
Just merge

**Pull request merged**